internal/xds/rbac/rbac_engine.go

```diff
@@ -112,6 +112,12 @@ func (cre *ChainEngine) IsAuthorized(ctx context.Context) error {
 	return nil
 }
 
+func (cre *ChainEngine) SetName(name string) {
```

Comment:
If we are to have such a public method, can we unify two cases by also using this in https://github.com/grpc/grpc-go/blob/b0842e59c6cfbbe728a01d8765b0cb6c378b93d6/authz/grpc_authz_server_interceptors.go? We then can remove the `policyName` passed into `NewChainEngine` in `internal/xds/rbac` and `newEngine` below.

Reply:
Making sure I'm interpreting right - you're suggesting removing `name` from `NewChainEngine` and `newEngine`, not setting the name at all there, then using the new `SetName` function in the same way after they are built?

I agree it's cleaner not to set the name two different ways

Reply:
Will change, I like that - makes more sense than having two different ways of setting the name

Reply:
PTAL